src/web/core/src/main/java/org/geoserver/web/data/layergroup/LayerGroupEditPage.java

```diff
@@ -34,38 +72,222 @@ public LayerGroupEditPage(PageParameters parameters) {
-                        if (!(c instanceof AbstractLink && "cancel".equals(c.getId()))) {
-                            c.setEnabled(false);
+    private void postInit() {
+        if (!isAuthenticatedAsAdmin() && !isNew) {
+            // global layer groups only editable by full admin
+            if (getPublishedInfo().getWorkspace() == null) {
+                // disable all form components but cancel
+                setInputEnabled(false);
+
+                info(new StringResourceModel("globalLayerGroupReadOnly", this, null).getString());
+            }
+        }
+    }
+            
+    public class LayerGroupTab extends PublishedEditTabPanel<LayerGroupInfo> {
+
+        private static final long serialVersionUID = 2192005814142588155L;
+
+        public LayerGroupTab(String id) {
+            super(id, myModel);
+            initUI();
+        }
+
+        private EnvelopePanel envelopePanel;       
+        protected RootLayerEntryPanel rootLayerPanel;    
+        
+        private void initUI() {
+
+            final WebMarkupContainer rootLayerPanelContainer = new WebMarkupContainer("rootLayerContainer");
+            rootLayerPanelContainer.setOutputMarkupId(true);
+            add(rootLayerPanelContainer);        
+            
+            rootLayerPanel = new RootLayerEntryPanel("rootLayer", getPublishedInfo().getWorkspace(), myModel);
+            rootLayerPanelContainer.add(rootLayerPanel);
+
+            updateRootLayerPanel(getPublishedInfo().getMode());
+            
+            TextField<String> name = new TextField<String>("name");
+            name.setRequired(true);
+            //JD: don't need this, this is validated at the catalog level
+            //name.add(new GroupNameValidator());
+            add(name);
+
+            final DropDownChoice<LayerGroupInfo.Mode> modeChoice = new DropDownChoice<LayerGroupInfo.Mode>("mode", new LayerGroupModeModel(), new LayerGroupModeChoiceRenderer());
+            modeChoice.setNullValid(false);
+            modeChoice.setRequired(true);
+            modeChoice.add(new OnChangeAjaxBehavior() {
+                private static final long serialVersionUID = 8819356789334465887L;
+
+                @Override
+                protected void onUpdate(AjaxRequestTarget target) {
+                    LayerGroupInfo.Mode mode = modeChoice.getModelObject();
+                    updateRootLayerPanel(mode);
+                    target.addComponent(rootLayerPanelContainer);
+                }
+            });
+            
+            add(modeChoice);
+            
+            add(new TextField<String>("title"));
+            add(new TextArea<String>("abstract"));
+            
+            DropDownChoice<WorkspaceInfo> wsChoice = 
+                    new DropDownChoice<WorkspaceInfo>("workspace", new WorkspacesModel(), 
+                            new WorkspaceChoiceRenderer());
+            wsChoice.setNullValid(true);
+            if (!isAuthenticatedAsAdmin()) {
```

Comment:
@aaime You are mistaken, there is no lost behaviour. All behaviour towards workspaces is left the same. See https://github.com/geoserver/geoserver/pull/1348/files#diff-eec1d0f43ca47e20dc28a41b97cd41beR213 and https://github.com/geoserver/geoserver/pull/1348/files#diff-eec1d0f43ca47e20dc28a41b97cd41beR79.

The purpose of this patch is to make the layergroup page more similar to the layer page, and of course share the necessary code between them. Contrary to what you are saying, the layer (resource) page was already using a single class both for new and well as existing layers. As I was porting the layergroup page to become part of the same structure I followed the same system. So I was actually adapting to existing code.
I could make a derived class for a new layergroup but I see little purpose since it only regards a tiny piece of custom code that can easily be solved with an if condition (considered the boolean isNew is there regardless).